docs/basic-features/data-fetching/getStaticPaths.md

```diff
@@ -0,0 +1,60 @@
+---
+description: Fetch data at build time with `getStaticProps` API reference.
+---
+
+# `getStaticPaths`
+
+If a page has [dynamic routes](/docs/routing/dynamic-routes.md) and uses `getStaticProps`, it needs to define a list of paths that have to be rendered to HTML at build time.
+
+When you export an `async` function called `getStaticPaths` (static generation) from a page that uses dynamic routes, Next.js will statically pre-render all the paths specified by `getStaticPaths`.
+
+```jsx
+export async function getStaticPaths() {
+  return {
+    paths: [
+      { params: { ... } }
+    ],
+    fallback: true // false or blocking
+  };
+}
+```
+
+The [`getStaticPaths` API reference](/docs/api-reference/data-fetching/getStaticPaths.md) covers all parameters and props that can be used with `getStaticPaths`.
+
+## When should I use `getStaticPaths`?
+
+You should use `getStaticPaths` if you’re statically pre-rendering pages that use dynamic routes.
```

Comment:
I think we can remove this and put it in the intro paragraph

Reply:
Yes please, would be useful to have this context upfront. Same for the "When should I use getStaticProps?"